src/pages/index.js

```diff
@@ -29,51 +43,124 @@ const renderCard = (item) => {
 };
 
 //create objects
+//api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "7ff811c0-9f09-42c2-975e-cca434c12992",
+    "content-type": "application/json",
+    "Access-Control-Allow-Origin": "*",
+  },
+});
+
 //profile section
-const userInformation = new UserInfo(".profile");
+const userInfo = new UserInfo(".profile");
+
+const profileSection = new Section(
+  {
+    renderer: () => {
+      api.getUserInformation().then(({ name, avatar, about, _id }) => {
+        userInfo.setAllInfo({
+          name: name,
+          avatar: avatar,
+          about: about,
+          _id: _id,
+        });
+      });
+    },
+  },
+  ".profile"
+);
 
 //cards section
 const cardList = new Section(
   {
-    data: initCards,
-    renderer: renderCard,
+    renderer: () => {
+      api.getInitialCards().then((items) => {
+        items.forEach((item) => {
+          renderCard(item);
+        });
+      });
+    },
   },
   ".page__cards"
 );
 
 //edit form
-const editFormValidator = new FormValidation("profile-edit-form", formData);
+const editFormValidator = new FormValidation("edit", formData);
 editFormValidator.enableValidation();
 
 const editPopup = new PopupWithForm(".modal_type_edit", (values) => {
-  userInformation.setUserInfo(values);
-  editFormValidator.disableSubmit();
-  editPopup.close();
+  editPopup.toggleSaving();
+  api.updateUserInformation(values).then((values) => {
+    userInfo.setUserInfo(values);
+    editPopup.close();
+    editPopup.toggleSaving();
```

Comment:
` editPopup.toggleSaving();`


You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.


**Please, correct it everywhere**